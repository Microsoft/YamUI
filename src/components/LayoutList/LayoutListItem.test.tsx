/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { LayoutListItem, LayoutListItemProps } from '.';

describe('<LayoutListItem />', () => {
  let component: ShallowWrapper<LayoutListItemProps>;

  describe('without props', () => {
    beforeEach(() => {
      component = shallow(<LayoutListItem />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<LayoutListItem className="TEST_CLASSNAME" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with child content', () => {
    beforeEach(() => {
      component = shallow(<LayoutListItem>list item content</LayoutListItem>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
