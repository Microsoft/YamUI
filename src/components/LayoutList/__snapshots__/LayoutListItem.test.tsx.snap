// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutListItem /> with additional className matches its snapshot 1`] = `
<li
  className="y-layoutList--item TEST_CLASSNAME"
/>
`;

exports[`<LayoutListItem /> with child content matches its snapshot 1`] = `
<li
  className="y-layoutList--item"
>
  list item content
</li>
`;

exports[`<LayoutListItem /> without props matches its snapshot 1`] = `
<li
  className="y-layoutList--item"
/>
`;
